SalesforceCommerceCloud/pwa-kit
# @W-12627216@ Mobile account menu a11y fixes

This is a follow up PR to #1884 

This PR contains a couple of a11y fixes for the mobile view's My Account page dropdown menu

* Using a header instead of `<p>` tag on the account drop down menu header
* Using box to indicate current selection in addition to color and the left border highlight
* Using underline to indicate hover in addition to color. This change also affects other menu-link items.
* Wrapping the menu items in a `<li>` tag
* Adjusted mobile view Footer heading to `<h1>` so screen readers do not think the footer is a sub section of the account page.

There is also some styling changes so that the header is better aligned with the menu items and to make all 'My Account' labels consistent.

Before:
![Screenshot 2024-10-07 at 5 03 24 PM](https://github.com/user-attachments/assets/3607b390-1719-4955-88eb-32be30616d7b)

After:
![Screenshot 2024-10-07 at 5 04 36 PM](https://github.com/user-attachments/assets/28bfd4ea-1e05-498b-afbf-4cc0e301ddc7)


Testing these changes:

- Pull down these changes and start the app
- Log in
- Size your window to trigger mobile view
- Expand the 'My Account' drop down menu
- Verify menu header is `<h2>` instead of `<p>`. Also verify that the text styling is the same as before
- Verify each menu item is wrapped in a `<li>`
- Verify that hovering over menu items now underlines the selection
- Click a menu item
- Verify the selected page is now indicated in the drop down menu
